platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/SupportMapFragment.java

```diff
@@ -25,10 +28,11 @@
  *
  * @see #getMapAsync(OnMapReadyCallback)
  */
-public class SupportMapFragment extends Fragment {
+public class SupportMapFragment extends Fragment implements OnMapReadyCallback {
 
+  private final List<OnMapReadyCallback> mapReadyCallbackList = new ArrayList<>();
```

Comment:
Since we don't have an API with a list of listeners for `OnMapReadyCallback` when instantiating it by hand I don't know why should we include it here. Was that a requested feature?

Reply:
Should have added that to the OP, but https://github.com/mapbox/mapbox-gl-native/issues/9923 is one. There is also this PR but it never got merged. Also there are users that have complained about this when using RxJava, they call getMapAsync multiple times and later as the default lifecycle.